src/fake-timers-src.js

```diff
@@ -1635,6 +1637,16 @@ function withGlobal(_global) {
             );
         }
 
+        if (
+            typeof stockDate === "function" &&
+            Math.abs(new Date().getTime() - new stockDate().getTime()) > 100
+        ) {
```

Comment:
I do not fancy the implicitness of this. It also won't cover a bunch of scenarios where we _do not_ fake `Date`. fake-timers support faking a single timer or many the APIs. See the `toFake` config option: https://github.com/sinonjs/fake-timers#var-clock--faketimersinstallconfig

I would much rather be specific in testing for a property on the various timers, along the lines of what [Luxon `DateTime.isDatetime(obj)`](https://moment.github.io/luxon/api-docs/index.html#datetime) is doing. This would only mean adding a boolean `isFakeTimer` to Date and the timers. And the test would be explicit!

Reply:
I noted that the faked timers can't be used unless `install` is first called, so I tried to put the property on the global only when `install` is called. I tried to leave the global object clean after `uninstall`.